pkg/registry/core/namespace/storage/storage.go

```diff
@@ -94,6 +95,12 @@ func (r *REST) NamespaceScoped() bool {
 	return r.store.NamespaceScoped()
 }
 
+var _ rest.SingularNameProvider = &REST{}
+
+func (r *REST) GetSingularName() string {
```

Comment:
this stands out.  Why does this one need it?

Reply:
because it doesn't inline r.store and inherit the default implementation

Reply:
it does not inherit default store implementation as said ^ https://github.com/kubernetes/kubernetes/blob/43a889fc65bc4919794ce9d43282fd2166856ad8/pkg/registry/core/namespace/storage/storage.go#L44